onadata/apps/api/tools.py

```diff
@@ -76,6 +77,22 @@ def _get_id_for_type(record, mongo_field):
         else mongo_str
 
 
+def create_xform_version(xform: XForm, user: User) -> XFormVersion:
+    """
+    Creates an XFormVersion object for the passed in XForm
+    """
+    if xform.version and not XFormVersion.objects.filter(
+            xform=xform, version=xform.version).first():
+        return XFormVersion.objects.create(
+            xform=xform,
+            xls=xform.xls,
+            json=xform.json,
+            version=xform.version,
+            created_by=user,
+            xml=xform.xml
+        )
```

Comment:
I think the norm is to create the record and have it fail on an `IntegrityError` exception if the record already exists, instead of checking before inserting. We are likely to have duplicate records when two separate process run at about the same time.